src/Microsoft.TemplateEngine.Edge/Constraints/WorkloadConstraintFactory.cs

```diff
@@ -0,0 +1,127 @@
+
+        public string Type => "workload";
+
+        public Task<ITemplateConstraint> CreateTemplateConstraintAsync(IEngineEnvironmentSettings environmentSettings, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            return Task.FromResult((ITemplateConstraint)new WorkloadConstraint(environmentSettings, this));
+        }
+
+        internal class WorkloadConstraint : ConstraintBase
+        {
+            private readonly HashSet<string> _installedWorkloads;
+            private readonly string _installedWorkloadsString;
+
+            internal WorkloadConstraint(IEngineEnvironmentSettings environmentSettings, ITemplateConstraintFactory factory)
+                : base(environmentSettings, factory)
+            {
+                IReadOnlyList<WorkloadInfo> workloads = ExtractWorkloadInfo(environmentSettings.Components.OfType<IWorkloadsInfoProvider>(), environmentSettings.Host.Logger);
+                _installedWorkloads = new HashSet<string>(workloads.Select(w => w.Id), StringComparer.InvariantCultureIgnoreCase);
+                _installedWorkloadsString = workloads.Select(w => $"{w.Id} \"{w.Description}\"").ToCsvString();
+            }
+
+            public override string DisplayName => "Workload";
+
+            protected override TemplateConstraintResult EvaluateInternal(string? args)
+            {
+                IReadOnlyList<string> supportedWorkloads = ParseArgs(args).ToList();
+
+                bool isSupportedWorkload = supportedWorkloads.Any(_installedWorkloads.Contains);
+
+                if (isSupportedWorkload)
+                {
+                    return TemplateConstraintResult.CreateAllowed(Type);
+                }
+
+                return TemplateConstraintResult.CreateRestricted(
+                    Type,
+                    string.Format(
+                        LocalizableStrings.WorkloadConstraint_Message_Restricted,
+                        string.Join(", ", supportedWorkloads),
+                        string.Join(", ", _installedWorkloadsString)),
+                    LocalizableStrings.Workload_Message_CTA);
+            }
+
+            //supported configuration:
+            // "args": "maui-mobile"  // workload expected
+            // "args": [ "maui-mobile", "maui-desktop" ] // any of workloads expected
+            private static IEnumerable<string> ParseArgs(string? args)
+            {
+                return args.ParseConstraintStrings();
+            }
+
+            private static IReadOnlyList<WorkloadInfo> ExtractWorkloadInfo(IEnumerable<IWorkloadsInfoProvider> workloadsInfoProviders, ILogger logger)
+            {
+                List<WorkloadInfo>? workloads = null;
+                List<Guid> previousComponentsGuids = new List<Guid>();
+
+                foreach (IWorkloadsInfoProvider workloadsInfoProvider in workloadsInfoProviders)
```

Comment:
Should we just accumulate all installed workflows? I think version might be important too (if available)

Reply:
Version seems to be part of the workload string id: https://github.com/dotnet/designs/blob/main/accepted/2020/workloads/workload-manifest.md#side-by-side-workload-pattern